rbac/migration_tool/utils.py

```diff
@@ -0,0 +1,30 @@
+"""Utilities for working with the relation API server."""
+import grpc
+from django.conf import settings
+from relations.v0 import common_pb2
+from relations.v0 import relation_tuples_pb2
+from relations.v0 import relation_tuples_pb2_grpc
+
+
+def create_relationship(resource_name, resource_id, subject_name, subject_id, relation):
+    """Create a relationship between a resource and a subject."""
+    return common_pb2.Relationship(
```

Comment:
There is support for validation at proto level - and thanks to that you can validate request before request is sent

There is example:
https://github.com/project-kessel/relations-client-python/blob/main/examples/client_validations.py#L75C20-L91C51
(Unfortunately you need to validate each part of request separately as is in example - validations did not work for whole request)

This could help catch error like empty data sooner.

Reply:
Updated